src/vivarium_csu_alzheimers/components/testing.py

```diff
@@ -80,14 +80,20 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
 
         # Update to reflect history
         event_time = pop_data.creation_time + get_timedelta_from_step_size(self.step_size)
-        bbbm_eligible_mask = self._get_bbbm_eligible_simulants(pop, event_time)
-        pop[COLUMNS.BBBM_TEST_DATE] = self._generate_bbbm_testing_history(
-            pop, bbbm_eligible_mask, event_time
-        )
-        # All previous BBBM tests are negative
-        pop.loc[bbbm_eligible_mask, COLUMNS.BBBM_TEST_RESULT] = BBBM_TEST_RESULTS.NEGATIVE
-        pop.loc[bbbm_eligible_mask, COLUMNS.BBBM_TEST_EVER_ELIGIBLE] = True
-        pop.loc[bbbm_eligible_mask, COLUMNS.TESTING_STATE] = TESTING_STATES.BBBM
+        if self.scenario.bbbm_testing:
+            bbbm_eligible_mask = self._get_bbbm_eligible_simulants(pop, event_time)
+            testing_rate = self._get_bbbm_testing_rate(event_time)
+            tested_mask = bbbm_eligible_mask & (
+                pop[COLUMNS.TESTING_PROPENSITY] < testing_rate
+            )
+            pop[COLUMNS.BBBM_TEST_DATE] = self._generate_bbbm_testing_history(
+                pop, tested_mask, event_time
+            )
+            # All previous BBBM tests are negative
+            pop.loc[bbbm_eligible_mask, COLUMNS.BBBM_TEST_RESULT] = BBBM_TEST_RESULTS.NEGATIVE
```

Comment:
I think this and the next two lines should use the `tested_mask` instead of the `bbbm_eligible_mask`